homeassistant/components/mqtt/__init__.py

```diff
@@ -365,14 +365,9 @@ async def async_setup_reload_service() -> None:
         async def _reload_config(call: ServiceCall) -> None:
             """Reload the platforms."""
             # Fetch updated manually configured items and validate
-            if (
-                config_yaml := await async_integration_yaml_config(hass, DOMAIN)
-            ) is None:
-                # Raise in case we have an invalid configuration
-                raise HomeAssistantError(
-                    "Error reloading manually configured MQTT items, "
-                    "check your configuration.yaml"
-                )
+            config_yaml = await async_integration_yaml_config(hass, DOMAIN, True)
+            if TYPE_CHECKING:
+                assert config_yaml is not None
```

Comment:
Don't do like this, instead add typing @overload variants of `async_integration_yaml_config` and `async_process_component_config` which return only `ConfigType` when `raise_on_failure` is `True`.